v2/apiserver/internal/core/projects.go

```diff
@@ -240,6 +243,36 @@ func (p *projectsService) Delete(ctx context.Context, id string) error {
 		return errors.Wrapf(err, "error retrieving project %q from store", id)
 	}
 
+	// Locate all events associated with this project
+	eventList, err := p.eventsStore.List(
+		ctx,
+		EventsSelector{ProjectID: id, WorkerPhases: WorkerPhasesAll()},
+		meta.ListOptions{},
+	)
+	if err != nil {
+		return errors.Wrapf(
+			err,
+			"error listing events associated with project %q from store",
+			id,
+		)
+	}
+
+	// Delete all events associated with this project
+	for _, event := range eventList.Items {
+		if err = p.eventsStore.Delete(ctx, event.ID); err != nil {
+			return errors.Wrapf(err, "error deleting event %q from store", id)
+		}
+
+		if err = p.substrate.DeleteWorkerAndJobs(ctx, project, event); err != nil {
+			return errors.Wrapf(
+				err,
+				"error deleting event %q worker and jobs from the substrate",
+				id,
+			)
+		}
```

Comment:
I think the call to `p.substrate.DeleteProject` on line 279 covers this. It delete's the project's namespace-- and that would take everything with it.

Reply:
Yes indeed, thank you for pointing this out.  Updated.